hstrat/phylogenetic_inference/pairwise/_calc_rank_of_mrca_bounds_between.py

```diff
@@ -128,13 +133,33 @@ def calc_rank_of_mrca_bounds_between(
             confidence_level_commonality=confidence_level,
             confidence_level_disparity=0.49,  # 0.49 i.e., definitive max
         )
-        assert (
-            first_disparity is not None
-            or first.GetNumStrataDeposited() == second.GetNumStrataDeposited()
-        )
+
+        # change in logic to handle surfaces potentially ignoring ranks
+        if isinstance(first, HereditaryStratigraphicColumn) and isinstance(
```

Comment:
Instead of putting the assert in an if statement, we should just add
`or isinstance(first, HereditaryStratigraphicSurface)`
`or isinstance(second, HereditaryStratigraphicsSurface)`

Reply:
Actually, I think the logic of having necessarily discordant ranks for different numbers of strata deposited still holds for surfaces (even if the most recent strata is not retained)